src/x509/cert.ts

```diff
@@ -146,6 +161,11 @@ export class x509Certificate {
     return ca;
   }
 
+  public extension(oid: string): x509Extension | undefined {
+    const ext = this.findExtension(oid);
+    return ext ? new x509Extension(ext) : undefined;
+  }
+
```

Comment:
Updating the `x509Certificate` class to support the retrieval of arbitrary extensions by OID in support of the OID verification checks (i.e. we need to be able to check that a certificate contains specific extensions with specific values).